j2cl-maven-plugin/src/main/java/com/vertispan/j2cl/mojo/TestMojo.java

```diff
@@ -312,7 +312,7 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             mainDep.setProject(main);
             testDeps.add(mainDep);
             test.setDependencies(testDeps);
-            test.setSourceRoots(new ArrayList<>(project.getTestCompileSourceRoots()));
+            test.setSourceRoots(project.getTestCompileSourceRoots().stream().filter(withTestSourcePathFilter()).collect(Collectors.toList()));
```

Comment:
Sorry, to clarify, I think it would be helpful to apply this filter not only to TestMojo, but always apply it in AbstractBuildMojo. This way it could in theory be possible to run a non-test goal using the test classes/classpath.